src/Teams/Data/Repositories/MultipleAnswerQuestionRepository.cs

```diff
@@ -9,14 +9,31 @@ namespace Teams.Data.Repositories
 {
     public class MultipleAnswerQuestionRepository : IMultipleAnswerQuestionRepository
     {
-        ApplicationDbContext _db;
-        public MultipleAnswerQuestionRepository(ApplicationDbContext db)
+        IApplicationDbContext _db;
+        public MultipleAnswerQuestionRepository(IApplicationDbContext db)
         {
             _db = db;
         }
         public MultipleAnswerQuestion PickById(Guid Id)
         {
             return _db.MultipleAnswerQuestions.Include(a => a.Answers).Single(q => q.Id == Id);
         }
+
+        public void AddQuestion(MultipleAnswerQuestion question)
+        {
+            _db.MultipleAnswerQuestions.Add(question);
+        }
+
+        public void DeleteQuestionOptionsIn_DB(MultipleAnswerQuestion question)
+        {
+            question.Answers.ToList().ForEach(o => _db.Entry(o).State = EntityState.Deleted);
+        }
+
+        public void UpdateQuestion(MultipleAnswerQuestion question)
+        {
+            _db.MultipleAnswerQuestions.Update(question);
+            _db.SaveChanges();
```

Comment:
Why method add question do not contain SaveChanges but this one do. Let's decide we call saveChanges in all write methods in repository or create separate one and call explicitly.